src/story.ts

```diff
@@ -8,24 +8,28 @@ type StoryReturnType<T> = T extends (storyFn: infer StoryFn, context: any) => an
     : never
   : never;
 
-export interface KindMeta<Component = unknown, Decorator = unknown> {
+type DefaultParameters = { [name: string]: unknown };
+
+type ParametersType<T extends { [P: string]: any }> = { [P in keyof T]: T[P] };
+
+export interface KindMeta<
+  Component = unknown,
+  Parameters = DefaultParameters,
+  Decorator = unknown
+> {
   id?: StoryId;
   title: StoryKind;
   component?: Component;
   subcomponents?: Record<string, Component>;
   decorators?: Decorator[];
-  parameters?: {
-    [name: string]: unknown;
-  };
+  parameters?: ParametersType<Parameters>;
 }
 
-export interface StoryMeta<Decorator = unknown> {
+export interface StoryMeta<Parameters = DefaultParameters, Decorator = unknown> {
   (): StoryReturnType<Decorator>;
   story?: {
     name?: StoryName;
     decorators?: Decorator[];
-    parameters?: {
-      [name: string]: unknown;
-    };
+    parameters?: ParametersType<Parameters>;
```

Comment:
We don't know the types of parameters since every addon has its own parameter schema. So I'd suggest this change is a step in the wrong direction -- making the typings more complex without any user benefit.

Reply:
I simply want to give to the user possibility define fully defined types for used addons. But I agree, this add a lot of work to define types for every addon. I revert this to as it has been before and replace `unknown` to `any` to less strict.

Reply:
Thanks @wkich, I think this is looking good. Would like to get some more eyes on this cc @tmeasday @ndelangen @gaetanmaisse

Reply:
I do like the generic version with `Parameters` but maybe with `any` instead of `unknown` in `DefaultParameters` in order to not be super strict with Storybook users. And it's meaningful only if Storybook addons are exporting a type corresponding to their parameters which isn't true for now, I think.

Reply:
@wKich / @shilman isn't there a "win win" if you make the type argument optional?

That way you can keep the existing behavior but allow people to override it to make it more specific. See here.

```ts
type ParametersType<T extends { [P: string]: any }> = { [P in keyof T]: T[P] };

type ParametersForgiving<T = Record<string, any>> = ParametersType<T>;

const looser: ParametersForgiving = {};

const stricter: ParametersForgiving<{param1: string, param2: boolean}> = {
    param1: "value1"
    // Property 'param2' is missing in type '{ param1: string; }' but required in type 'ParametersType<{ param1: string; param2: boolean; }>'.
};
```

Reply:
@dgreene1, yeah, this's exactly that I did in commit https://github.com/storybookjs/csf/pull/5/commits/48e6e3353f79074065a81524e6671866794485b7

@shilman, I think generic version is more popular. It has fallback, so if some addons don't have typings for now, it's ok, and all works fine.